### PR TITLE
Finished the porting of t_pubsub_thr

### PR DESCRIPTION
**Pull request by YuanYuYuan**
* Pass the compilation: `cargo build --release --bin t_pubsub_thr`.
* Test it with `nice -10 taskset -c 0,1 ./target/release/t_pubsub_thr --payload 8 --mode peer --name test --scenario test --print` .

### CODE REVIEW

File: throughput/src/bin/t_pubsub_thr.rs [hunk 1]
```diff
@@ -108,24 +112,38 @@ impl TransportPeerEventHandler for MyMH {
     }
 }
 
-#[derive(Debug, StructOpt)]
+#[derive(Debug, Parser)]
 #[structopt(name = "s_pubsub_thr")]
 struct Opt {
-    #[structopt(short = "l", long = "listen")]
+    /// which endpoints to listen on. `zenohd` will add `tcp/0.0.0.0:7447` to these locators if left empty,
```
Comment (william-wyliang) [inline throughput/src/bin/t_pubsub_thr.rs]: Comment only. Is the description of "`zenohd` will add `tcp/0.0.0.0:7447` to these locators if left empty" necessary? Or, say, is it only related to `zenohd`?

**Review by william-wyliang — APPROVED** [review]
One inline comment for discussion only.

File: throughput/src/bin/t_pubsub_thr.rs (hunk 1 above)
Comment (YuanYuYuan) [reply throughput/src/bin/t_pubsub_thr.rs]: I just copied the documentation from zenoh source code. It may be a good idea to remove them. Thanks!

Comment (YuanYuYuan) [reply throughput/src/bin/t_pubsub_thr.rs]: @eddie9712 I just made a new PR, can you help me review it and merge it?

Comment (william-wyliang) [reply throughput/src/bin/t_pubsub_thr.rs]: I'm confident with the change @YuanYuYuan made. To make the PR done asap without further waiting, I'll do the merge.

**Pull request merged**